File: 02 FormatString/java/FormatString.java
import java.text.MessageFormat;

public class FormatString {
  public static void main(String[] args) {
    var language = "C#";
    var price = 24.99F;
    byte quotas = 20;

    var concatenation = "Course: " + language + " Programming Fundamentals \n" + "Price: $" + price + " \n" + "Quotas: "
        + quotas + " \n";

    var formatString = String.format("Course: %s Programming Fundamentals \nPrice: $%f \nQuotas: %d", language, price,
        quotas);

    var builder = new StringBuilder();
    builder.append("Course: ").append(language).append(" Programming Fundamentals \n").append("Price: $").append(price)
        .append(" \n").append("Quotas: ").append(quotas).append(" \n");
    var builderString = builder.toString();

    var messageFormat = MessageFormat.format("Course: {0} Programming Fundamentals \nPrice: ${1} \nQuotas: {2}",
        language, price, quotas);
  }
}

File: 02 FormatString/python/format_string.py
language = "C#"
price = 24.99
quotas = 20

concatenation = "Course: " + language + " Programming Fundamentals \n" + "Price: $" + str(price) + " \n" + "Quotas: " + str(quotas) + " \n"

stringFormat = "Course: {2} Programming Fundamentals \nPrice: ${1} \nQuotas: {0}".format(language, price, quotas)

stringFormat2 = "Course: {pepito} Programming Fundamentals \nPrice: ${price} \nQuotas: {quotas}".format(pepito = language, price = price, quotas = quotas)

stringFormat3 = "Course: {} Programming Fundamentals \nPrice: ${} \nQuotas: {}".format(language, price, quotas)

interpolation = f"Course: {language} Programming Fundamentals \nPrice: ${price} \nQuotas: {quotas}"

stringFormat% = "Course: %s Programming Fundamentals \nPrice: $%d \nQuotas: %d"%(language, price, quotas)